packages/iov-bns/src/types.ts

```diff
@@ -23,6 +23,89 @@ export interface CashConfiguration {
   readonly minimalFee: Amount;
 }
 
+// Governance
+
+export interface Elector {
+  readonly address: Address;
+  /** The voting weight of this elector. Max value is 65535 (2^16-1). */
+  readonly weight: number;
+}
+
+export interface Electorate {
+  readonly id: number;
+  readonly version: number;
+  readonly admin: Address;
+  readonly title: string;
+  /** An unordered map from address to elector */
+  readonly electors: { readonly [index: string]: Elector };
+  /** Sum of all electors' weights */
+  readonly totalWeight: number;
+}
+
+export interface Fraction {
+  readonly numerator: number;
+  readonly denominator: number;
+}
+
+export interface ElectionRule {
+  readonly id: Uint8Array;
+  readonly version: number;
+  readonly admin: Address;
+  /** The electorate used by this (ID without version, as changing electorate changes the rule) */
+  readonly electorateId: number;
+  readonly title: string;
+  /** Voting period in seconds */
+  readonly votingPeriod: number;
+  readonly threshold: Fraction;
+  readonly quorum: Fraction | null;
+}
+
+export interface VersionedId {
+  readonly id: Uint8Array;
+  readonly version: number;
+}
+
+export enum ProposalExecutorResult {
+  NotRun,
+  Succeeded,
+  Failed,
+}
+
+export enum ProposalResult {
+  Undefined,
+  Accepted,
+  Rejected,
+}
+
+export enum ProposalStatus {
+  Submitted,
+  Closed,
+  Withdrawn,
+}
+
+/** Union type for possible options */
+export type ProposalOption = string;
```

Comment:
Union type?

Reply:
`string | never` ;) This is just the beginning of a type that will later on be able to express all the different actions that can be executed when a proposal is accepted.